### PR TITLE
HW-Hibernate-5OfJuly-20-21

### PR DESCRIPTION
**Pull request by ViktoriiaSvidovska**

### CODE REVIEW

File: src/main/java/mate/academy/dao/CinemaHallDao.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+public interface CinemaHallDao {
+
+    CinemaHall add(CinemaHall cinemaHall);
+
+    Optional<CinemaHall> get(Long id);
+
+    List<CinemaHall> getAll();
+
+}
```
Comment (Rommelua) [inline src/main/java/mate/academy/dao/CinemaHallDao.java:15]:
```suggestion
public interface CinemaHallDao {
    CinemaHall add(CinemaHall cinemaHall);

    Optional<CinemaHall> get(Long id);

    List<CinemaHall> getAll();
}
```

File: src/main/java/mate/academy/dao/MovieDao.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import mate.academy.model.Movie;
 
 public interface MovieDao {
+
     Movie add(Movie movie);
```
Comment (Rommelua) [inline src/main/java/mate/academy/dao/MovieDao.java:9]:
```suggestion
public interface MovieDao {
    Movie add(Movie movie);
```

File: src/main/java/mate/academy/dao/MovieSessionDao.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package mate.academy.dao;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.model.MovieSession;
+
+public interface MovieSessionDao {
+
+    MovieSession add(MovieSession movieSession);
```
Comment (Rommelua) [inline src/main/java/mate/academy/dao/MovieSessionDao.java:10]:
```suggestion
public interface MovieSessionDao {
    MovieSession add(MovieSession movieSession);
```

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+
+    @Override
```
Comment (Rommelua) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:20]:
```suggestion
public class MovieSessionDaoImpl implements MovieSessionDao {
    @Override
```

File: src/main/java/mate/academy/model/CinemaHall.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "cinema_halls")
+public class CinemaHall implements Cloneable {
+
+    @Id
```
Comment (Rommelua) [inline src/main/java/mate/academy/model/CinemaHall.java:13]:
```suggestion
public class CinemaHall {
    @Id
```

**Review by Rommelua — CHANGES_REQUESTED** [review]
Do not begin class with empty line
Remove Clonable interface everywhere

File: src/main/java/mate/academy/dao/CinemaHallDao.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import java.util.Optional;
+import mate.academy.model.CinemaHall;
+
+public interface CinemaHallDao {
+    CinemaHall add(CinemaHall cinemaHall);
+
+    Optional<CinemaHall> get(Long id);
+
+    List<CinemaHall> getAll();
+
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/dao/CinemaHallDao.java:13]:
```suggestion
```

At the end, spaces must also be removed :)

File: src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import javax.persistence.EntityNotFoundException;
+import mate.academy.dao.MovieSessionDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.service.MovieSessionService;
+
+@Service
+public class MovieSessionServiceImpl implements MovieSessionService {
+
```
Comment (kozhukhovsky) [inline src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java:14]:
```suggestion
```

File: src/main/resources/hibernate.cfg.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+        <property name="hibernate.connection.username">root</property>
+        <property name="hibernate.connection.password">mateacademy1</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+        <property name="show_sql">true</property>
+
+        <mapping class="mate.academy.model.CinemaHall"/>
+        <mapping class="mate.academy.model.Movie"/>
+        <mapping class="mate.academy.model.MovieSession"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (kozhukhovsky) [inline src/main/resources/hibernate.cfg.xml:18]:
```suggestion
</hibernate-configuration>

```

**Review by kozhukhovsky — APPROVED** [review]

**Pull request closed without merging**